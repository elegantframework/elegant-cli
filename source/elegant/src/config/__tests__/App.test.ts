import AppConfig from './../App';

describe('App Config', () => {
    // we need to mock out process.env functionality to test when an .env file isn't loaded
    const env = process.env;

    beforeEach(() => {
        jest.resetModules()
        process.env = { ...env }
    });

    afterEach(() => {
        process.env = env
    });

    it('returns an application name when an environment variable is passed',() => {

      let result = AppConfig().name;

      expect(result).toBe("Elegant - The unit testing application.");
    });

    it('returns the base config name when no env file is present',() => {

        process.env.NEXT_PUBLIC_APP_NAME = undefined;
        
        let result = AppConfig().name;
  
        expect(result).toBe("Elegant");
    });

    it('matches the existing snapshot when an env var is set',() => {
        expect(AppConfig()).toMatchSnapshot();
    });
});