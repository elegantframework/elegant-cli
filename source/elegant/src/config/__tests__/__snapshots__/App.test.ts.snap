// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Config matches the existing snapshot when an env var is set 1`] = `
{
  "name": "Elegant - The unit testing application.",
}
`;
